crates/wasmparser/src/binary_reader.rs

```diff
@@ -73,7 +73,7 @@ impl Range {
 }
 
 /// A binary reader of the WebAssembly structures and types.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Hash)]
```

Comment:
I wonder how/where `Hash` will be used -- `buffer` can be long.

Reply:
FWIW this is [used here](https://github.com/bytecodealliance/wasmtime/pull/2059/files#diff-70306a13ab1567aab84c018246072737R523) where we has the contents of functions for the internal cache.